File: code-coaches/hard/its-a-sign/its_a_sign.rb
def is_palindrome(word)
  word == word.reverse
end

def contains_palindrome(words)
  words.each do |word|
    return true if is_palindrome(word)
  end
  false
end

def main
  words = []
  4.times do
    words << gets.chomp
  end

  if contains_palindrome(words)
    puts "Open"
  else
    puts "Trash"
  end
end

if __FILE__ == $0
  main
end

File: code-coaches/hard/its-a-sign/its_a_sign.py
def is_palindrome(word: str) -> bool:
    return word == word[::-1]


def contains_palindrome(words: list[str]) -> bool:
    for word in words:
        if is_palindrome(word):
            return True

    return False


def main():
    words = [input() for _ in range(4)]

    if contains_palindrome(words):
        print("Open")
    else:
        print("Trash")


if __name__ == "__main__":
    main()
